### PR TITLE
Bundle included metrics

### LINKED ISSUE #3328: Add metrics of Bundles included in the block
Currently there is no metrics of how many bundles was included in the block.

### PR DESCRIPTION
**Pull request by LukaszRozmej**
Closes #3328

## Changes:
Add bundle included metrics

## Types of changes

What types of changes does your code introduce?

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Other (please describe): 

## Testing
**Requires testing**

- [x] Yes
- [ ] No

**In case you checked yes, did you write tests??**

- [x] Yes
- [ ] No

### CODE REVIEW

**Review by marcindsobczak — APPROVED** [review]

**Review by tkstanczak — CHANGES_REQUESTED** [review]
do not analyze this way, just add++ when included

File: src/Nethermind/Nethermind.Mev.Test/MevRpcModuleTests.TestMevRpcBlockchain.cs [hunk 1]
```diff
@@ -98,7 +98,9 @@ public TestMevRpcBlockchain(int maxMergedBundles, UInt256? initialBaseFeePerGas)
             protected override IBlockProducer CreateTestBlockProducer(TxPoolTxSource txPoolTxSource, ISealer sealer, ITransactionComparerProvider transactionComparerProvider)
             {
                 MiningConfig miningConfig = new() {MinGasPrice = UInt256.One};
-                
+
+                MevBlockProducerTransactionsExecutorFactory mevExecutorFactory = new(SpecProvider, LogManager);
```
Comment (tkstanczak) [inline src/Nethermind/Nethermind.Mev.Test/MevRpcModuleTests.TestMevRpcBlockchain.cs:102]: this looks already too complicated...

**Review by tkstanczak — CHANGES_REQUESTED** [review]
too complex for the task

File: src/Nethermind/Nethermind.Mev/MevBlockProducer.cs [hunk 2]
```diff
@@ -38,20 +38,24 @@ private class MevBestBlockPicker : IBestBlockPicker
...
                 UInt256 maxBalance = UInt256.Zero;
+                int bundlesAdded = 0;
                 foreach ((Block? Block, MevBlockProducerInfo BlockProducerInfo) context in blocks)
                 {
                     if (context.Block is not null)
                     {
-                        BeneficiaryTracer beneficiaryTracer = context.BlockProducerInfo.BeneficiaryTracer;
-                        UInt256 balance = beneficiaryTracer.BeneficiaryBalance;
+                        MevBlockProducerInfo mevBlockProducerInfo = context.BlockProducerInfo;
+                        UInt256 balance = mevBlockProducerInfo.BeneficiaryTracer.BeneficiaryBalance;
```
Comment (tkstanczak) [inline src/Nethermind/Nethermind.Mev/MevBlockProducer.cs:47]: law of demeter broken

File: src/Nethermind/Nethermind.Mev/MevBlockProducer.cs [hunk 3]
```diff
@@ -38,20 +38,24 @@ private class MevBestBlockPicker : IBestBlockPicker
...
                         if (balance > maxBalance || best is null)
                         {
                             best = context.Block;
                             maxBalance = balance;
+                            bundlesAdded = mevBlockProducerInfo.BundlesIncludedTracer.BundlesIncluded;
```
Comment (tkstanczak) [inline src/Nethermind/Nethermind.Mev/MevBlockProducer.cs:52]: law of demeter broken

File: src/Nethermind/Nethermind.Mev/MevBlockProducerTransactionsExecutorFactory.cs [hunk 4]
```diff
@@ -34,6 +34,8 @@ public MevBlockProducerTransactionsExecutorFactory(ISpecProvider specProvider, I
         }
 
         public IBlockProcessor.IBlockTransactionsExecutor Create(ReadOnlyTxProcessingEnv readOnlyTxProcessingEnv) => 
-            new MevBlockProductionTransactionsExecutor(readOnlyTxProcessingEnv, _specProvider, _logManager);
+            LastExecutor = new MevBlockProductionTransactionsExecutor(readOnlyTxProcessingEnv, _specProvider, _logManager);
```
Comment (tkstanczak) [inline src/Nethermind/Nethermind.Mev/MevBlockProducerTransactionsExecutorFactory.cs:37]: factory should be stateless if possible, looks like much complexity added

**Pull request closed without merging**